### PR TITLE
Ability to create Java host buffers from memory-mapped files [skip ci]

### PR DESCRIPTION
**Pull request by jlowe**
This adds the ability to create a host memory buffer that is backed by a memory-mapped file.  Reading and writing to the file are supported via memory operations.

In addition this exposes some methods to allow applications to obtain the addresses of memory buffers and build columns from a series of buffers (i.e.: data, validity, offsets).

### CODE REVIEW

**Comment by jlowe** [comment]
One reservation I have is exposing the `ColumnVector` constructor that takes an `Optional<Long>` for the null count.  I'd much rather just take a long here and use the existing `UNKNOWN_NULL_COUNT` constant to indicate the null count is unknown.  Needing to deal with the `Optional` is cumbersome in practice, and I'm not sure it's adding any value.  @revans2 any thoughts on this?

File: java/src/main/java/ai/rapids/cudf/HostMemoryBuffer.java [hunk 1]
```diff
@@ -109,6 +139,38 @@ public static HostMemoryBuffer allocate(long bytes) {
...
 
+  /**
+   * Create a host buffer that is memory-mapped to a file.
+   * @param path path to the file to map into host memory
+   * @param mode mapping type
+   * @param offset file offset where the map will start
+   * @param length the number of bytes to map
+   * @return file-mapped buffer
+   */
+  public static HostMemoryBuffer mapFile(String path, MapMode mode,
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/HostMemoryBuffer.java]: nit: I personally would prefer to see path be a `File` or a `Path` just so it is 100% clear what this is mapping in, and it is not a URL or some kind.

**Comment by revans2** [comment]
> One reservation I have is exposing the `ColumnVector` constructor that takes an `Optional<Long>` for the null count. I'd much rather just take a long here and use the existing `UNKNOWN_NULL_COUNT` constant to indicate the null count is unknown. Needing to deal with the `Optional` is cumbersome in practice, and I'm not sure it's adding any value. @revans2 any thoughts on this?

I personally would prefer to keep the null count as Optional because that is much more of a java way to do it.  But it is not that big of a deal in my opinion.  The issue is that the more we expose of the internals, the more customers are going to rely on the underlying implementation in native cudf which does change regularly.  This means they have to be more careful about how it is used and we might want to add some more documentation that this can and will change between versions so don't rely on it being the same.

**Comment by jlowe** [comment]
Thanks for the review!  I updated `mapFile` to use `File` since other parts of the API were already using it.  I left the use of Optional as-is.

**Comment by jlowe** [comment]
@kkraus14 when you get a chance, this is ready to be merged.  Thanks!

**Pull request merged**